aws/resource_aws_route53_health_check.go

```diff
@@ -32,6 +32,15 @@ func resourceAwsRoute53HealthCheck() *schema.Resource {
 				StateFunc: func(val interface{}) string {
 					return strings.ToUpper(val.(string))
 				},
+				ValidateFunc: validation.StringInSlice([]string{
+					route53.HealthCheckTypeCalculated,
+					route53.HealthCheckTypeCloudwatchMetric,
+					route53.HealthCheckTypeHttp,
+					route53.HealthCheckTypeHttpStrMatch,
+					route53.HealthCheckTypeHttps,
+					route53.HealthCheckTypeHttpsStrMatch,
+					route53.HealthCheckTypeTcp,
+				}, true),
```

Comment:
is there a reason this is case-insensitive?

Reply:
I worry because the `StateFunc` right above is explicitly converting things to uppercase. My guess is that the API accepts both casings, so to keep that support for existing configurations, turned off case sensitivity here for validation.